src/nvim/eval/funcs.c

```diff
@@ -791,22 +791,29 @@ static void f_chansend(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
 
   ptrdiff_t input_len = 0;
   char *input = NULL;
+  uint64_t id = (uint64_t)argvars[0].vval.v_number;
+#ifdef UNIX
+  bool crlf = false;
+#else
+  Channel *chan = find_channel(id);
+  bool crlf = (chan != NULL && chan->term) ? true: false;
+#endif
```

Comment:
out of scope for this PR, but I wonder if given https://github.com/neovim/neovim/issues/16957 and https://github.com/neovim/neovim/issues/6751 we might need to check 'shell' or something to decide chansend() line endings.

Reply:
I'll try manual test in mintty shell.

Reply:
Tried in MSys2 and MSys2/MinGW64 shells in mintty shell. Works. But that's because the shells tested all require CRLF line-ending. So you're right. We need a mapping table or similar to decide based on `shell`.

Reply:
on Windows, all shells should be able to handle CRLF. This was more of a "nice to have" idea. Should wait until we have an actual use case.